### PR TITLE
Go: Add support for provenance pretty-printing

### PR DESCRIPTION
**Pull request by owen-mc**
For qlref tests this uses the new  ability to use a post-processing query.

### CODE REVIEW

File: go/ql/test/library-tests/semmle/go/dataflow/ThreatModels/threat-models-flowtest2.expected [hunk 1]
```diff
@@ -1,7 +1,13 @@
+models
+| 1 | Source: github.com/nonexistent/sources; ; false; ExecuteQuery; ; ; ReturnValue; database; manual |
+| 1 | Summary: archive/tar; ; false; NewReader; ; ; Argument[0]; ReturnValue; taint; manual |
```
Comment (owen-mc) [inline go/ql/test/library-tests/semmle/go/dataflow/ThreatModels/threat-models-flowtest2.expected:3]: number 1 repeated twice 😕

File: go/ql/test/library-tests/semmle/go/dataflow/ThreatModels/threat-models-flowtest3.expected [hunk 2]
```diff
@@ -1,9 +1,19 @@
+models
+| 1 | Source: github.com/nonexistent/sources; ; false; ExecuteQuery; ; ; ReturnValue; database; manual |
+| 1 | Summary: archive/tar; ; false; NewReader; ; ; Argument[0]; ReturnValue; taint; manual |
+| 2 | Source: github.com/nonexistent/sources; ; false; ReadEnvironment; ; ; ReturnValue; environment; manual |
+| 2 | Summary: archive/tar; Header; true; FileInfo; ; ; Argument[receiver]; ReturnValue; taint; manual |
+| 3 | Source: github.com/nonexistent/sources; ; false; GetCliArg; ; ; ReturnValue; commandargs; manual |
+| 3 | Summary: archive/tar; Writer; true; WriteHeader; ; ; Argument[0]; Argument[receiver]; taint; manual |
```
Comment (owen-mc) [inline go/ql/test/library-tests/semmle/go/dataflow/ThreatModels/threat-models-flowtest3.expected:7]: Lots of repeats

**Comment by owen-mc** [comment]
6 tests failed with the following output:
```
Postprocess query result is ambiguous for column 1 of line 0 in output grid models
[74/523 comp 274ms eval 927ms] FAILED(EXECUTION) /home/runner/work/codeql/codeql/go/ql/test/query-tests/Security/CWE-347/MissingJwtSignatureCheck.qlref
```

**Comment by owen-mc** [comment]
I rebased on top of https://github.com/github/codeql/pull/17072, which changed the model numbering. I also added a commit reverting the post-processing for 6 tests which currently hit a bug in the post-processing implementation. When the bug is fixed I will re-convert them, but in the meantime it seems best to merge post-processing for all of the other tests.

File: go/ql/test/library-tests/semmle/go/dataflow/ThreatModels/threat-models-flowtest2.expected (hunk 1 above)
Comment (owen-mc) [reply go/ql/test/library-tests/semmle/go/dataflow/ThreatModels/threat-models-flowtest2.expected:3]: The repeats are caused by a bug in the post-processing procedure. We will accept them for now and aim to fix the bug soon.

**Review by smowton — APPROVED** [review]
Looks plausible except for the duplicate numbering problem already noted

**Pull request merged**